code/game/objects/items/devices/oxycandle.dm

```diff
@@ -0,0 +1,72 @@
+/obj/item/device/oxycandle
+	name = "Oxygen Candle"
+	desc = "Oxygen candles are flare-like steel tubes of sodium chlorate that produces oxygen through chemical process."
+	icon = 'icons/obj/candle.dmi'
+	icon_state = "candle1"
+	w_class = ITEMSIZE_SMALL // Should fit into internal's box or maybe pocket
+	var/target_pressure = ONE_ATMOSPHERE
+	var/datum/gas_mixture/air_contents = null
+	var/volume = 5500 //One tile has 2500 volume of air, so two tiles plus a bit extra
+	var/on = 0
+	var/activation_sound = 'sound/items/flare.ogg'
+	light_color = LIGHT_COLOR_FLARE
+	light_wedge = LIGHT_OMNI
```

Comment:
Omni is the default, you don't need to specify it.

Reply:
Alright